authorizer-app/src/app/components/rest-source-authorization/update-rest-source-user.component.html

```diff
@@ -33,9 +33,9 @@ <h3 *ngIf="isEditing">Edit details for User</h3>
 
   <div class="form-group" *ngIf="restSourceUser.projectId">
     <label for="userId">User ID:</label>
-    <select class="form-control" id="userId" [(ngModel)]="restSourceUser.userId" [disabled]="isEditing">
+    <select class="form-control" id="userId" [(ngModel)]="restSourceUser.externalId ? restSourceUser.externalId : restSourceUser.userId" #userId="ngModel" [disabled]="isEditing">
       <option [selected]="true" value="">Select a user</option>
-      <option [value]="user.id" *ngFor="let user of subjects">{{user.id}}</option>
+      <option *ngFor="let user of subjects" [value]="user.externalId ? user.externalId : user.id">{{user.externalId? user.externalId : user.id}}</option>
```

Comment:
Note that the value of an option should always be the `userId`, since the backend only processes that. Preferably, this list is sorted in the following way

1. items with external ID, sorted by externalId, ASC, perhaps again with displayed as `externalId (userId | slice:0:8)`
2. items without external ID, sorted by userId, ASC

To make this a bit more user friendly, could this a combobox, dropdown with text box instead? https://material.angular.io/components/autocomplete/overview
If possible the filter would select both from userId and externalId.